packages/libs/user-datasets/src/lib/Components/UploadForm.tsx

```diff
@@ -449,6 +501,287 @@ function UploadForm({
             onChange={setDescription}
           />
         </div>
+        <div className={'formSection'}>
+          <details className={cx('--AdditionalDetails')}>
+            <summary>Additional Details</summary>
+            <div className="additionalDetailsFormSection additionalDetailsFormSection--data-set-publications">
+              <FieldLabel htmlFor="data-set-publications" required={false}>
+                Publications (Optional)
+              </FieldLabel>
+              {publications.map((publication, index) => {
+                const updatePublicationsObject = createNestedInputUpdater({
+                  nestedInputObject: publications,
+                  index,
+                });
+                return (
+                  <PublicationInput
+                    n={index}
+                    pubMedId={publication.pubMedId}
+                    onAddPubmedId={(value: string) => {
+                      const updatedPublications = updatePublicationsObject(
+                        value,
+                        'pubMedId'
+                      ) as UserDatasetPublication[];
```

Comment:
If you use the generic <T> type and function above then all the `as XXXXX[]` assertions here and below are not necessary :-)

```suggestion
                      const updatedPublications = updatePublicationsObject(
                        value,
                        'pubMedId'
                      );
```